lib/blob/blobstore.c

```diff
@@ -424,6 +424,16 @@ blob_back_bs_destroy(struct spdk_blob *blob)
 	blob->back_bs_dev = NULL;
 }
 
+static void
+bs_snapshot_free_clusters_checksums(struct spdk_blob *blob)
```

Comment:
I am thinking, if we can retain the checksum if the corresponding cluster is unchanged (after a snapshot deletion & merge).

For example, :
1. The parent snapshot snap1 is being removed and its data will be merged into the child snapshot snap2
```
snap1:   AB_
   |
  V
snap2:  _CD
```
2. After the data merge, the snap2 becomes the status below. This means that we don't need to erase the checksum result of the last 2 clusters. Moreover, if cluster A (which belongs to snap1) already has its checksum result, it will be nice for snap2 to take it directly.
```
snap2:  ACD
```


Anyway, this is a nice-to-have improvement.

Reply:
Nice idea, it could be a good improvement.
The problem wit this solution is that operations like lvol deletion and inflation will have to calculate the checksum of all the merged clusters: this operation could take a while and so we will significantly change the behavior of these APIs. Instead, if we don't calculate immediately the checksum of merged cluster, we will leave the checksum array in an inconsistent status.

Reply:
Can we handle the cluster checksum list merge with a separate buffer & thread? Before we get the final result, we can leave the merged snapshot with empty checksum (and calculation blocking).

Reply:
I mean, if merging the cluster checksum lists takes time, we can leave the checksum empty for the merged snapshot first. After successfully handling the checksum list in separate buffers & threads, we can put the list in the new snapshot meta info.

Reply:
Ok, you mean execute the merge in background after having completed the delete/inflate.
It can be a future improvement.

Reply:
Same, would you create a GitHub ticket to track this improvement idea?

Reply:
Done: https://github.com/longhorn/longhorn/issues/10870